signetdev/host/signetdev_emulate.c

```diff
@@ -69,6 +69,9 @@ static void enter_state(int state)
 	g_deviceState.n_progress_components = 0;
 }
 
+// disable warning as this function is linked to but not used in signet-client code
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
```

Comment:
I like to avoid pragma's in favor of language extensions. If we need this lets use the "_Pragma" operator.

What would be better is to identify in what conditions this function doesn't need to be built and just not build it. If it's an obsolete function lets just remove it.

Reply:
ok, done.

No, it's not obsolete, the firmware uses it uses it , but the client code does not.  Maybe I can find something appropriate to us in #ifdef

Reply:
Actually I'm pretty sure it is obsolete. It's a static function and there are no references in this file.  At this point "enter_state()" is doing its job. This file is in the "host" folder so it is not part of the firmware. I can see how you might have got confused since I used firmware code as a starting point.